app/components/TaskView.vue

```diff
@@ -311,6 +369,36 @@ const openTaskInNewTab = () => {
   window.open(`/channel/${props.cid}/task/${props.taskId}`, '_blank')
 }
 
+const onFileUpload = async (files) => {
+  if (!files.length) return
+
+  for (const file of files) {
+    if (file.size > 10 * 1024 * 1024) { 
+      console.error('File too large')
+      // toast.add({ title: 'File too large', description: `${file.name} exceeds 10MB limit`, color: 'red' })
+      continue
+    }
+
+    try {
+      const response = await taskStore.uploadTaskAttachment(props.cid, props.taskId, file)
+      attachments.value.push(response)
+      // toast.add({ title: 'Upload successful', description: `${file.name} uploaded`, color: 'green' })
+    } catch (error) {
+      console.error('Error uploading file:', error)
+      // toast.add({ title: 'Upload failed', description: `${file.name} could not be uploaded`, color: 'red' })
+    }
+  }
+}
+
+const downloadAttachment = async (attachmentId) => {
+  await taskStore.downloadTaskAttachment(props.cid, props.taskId, attachmentId)
```

Comment:
Make sure to show spinner inside attachment UI and handle the error too. Remove async here.